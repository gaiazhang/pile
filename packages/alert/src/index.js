import Alert from './components/alert';
import alertBox from './components/alertWarp';

const { open, close } = alertBox;
Alert.open = open;
Alert.close = close;
export default Alert;
