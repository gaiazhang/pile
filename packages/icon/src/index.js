import * as React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import { compose, prefixClsProperty, sizeProperty, sizes } from '@pile/shared';

const Icon = ({ prefixCls, type, size, className, ...props }) => {
  const cls = classNames({
    [`${prefixCls}-icon-${type}`]: true,
    [className]: className,
    [size]: size,
  });

  return <i className={cls} {...props} />;
};

Icon.propTypes = {
  type: PropTypes.string.isRequired,
};

const enhance = compose(
  sizeProperty([sizes.SMALL, sizes.LARGE]),
  prefixClsProperty
);

export default enhance(Icon);
