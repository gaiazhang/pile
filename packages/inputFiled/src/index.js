import * as React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { prefixClsProperty } from '@pile/shared';
import InputLabel from './inputlabel';
import InputContent from './inputContent';

const InputFiled = ({ prefixCls, children, ...props }) => {
  const wrapCls = classNames({
    [`${prefixCls}-input-item`]: true,
  });

  return (
    <div className={wrapCls}>
      <InputLabel labeltext={children} />
      <InputContent {...props} />
    </div>
  );
};

InputFiled.propTypes = {
  type: PropTypes.string,
  children: PropTypes.string,
  placeholder: PropTypes.string,
  clearable: PropTypes.bool,
  autoFocus: PropTypes.bool,
  disabled: PropTypes.bool,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
};

InputFiled.defaultProps = {
  type: 'text',
  children: '',
  placeholder: '',
  clearable: false,
  autoFocus: false,
  disabled: false,
  onFocus() {},
  onBlur() {},
  onChange() {},
};

export default prefixClsProperty(InputFiled);
