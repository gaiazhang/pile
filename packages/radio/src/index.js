import Radio from './radio';
import RadioGroup from './radioGroup';

Radio.RadioGroup = RadioGroup

export default Radio
// export { default as Radio } from './radio';
