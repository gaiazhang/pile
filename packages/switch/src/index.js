import classnames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';
import { prefixClsProperty } from '@pile/shared';

class Switch extends React.Component {
  constructor(props) {
    super(props);
    const { checked, disabled } = props;

    this.onClick = this.onClick.bind(this);
    this.onKeyPress = this.onKeyPress.bind(this);
    this.state = {
      checked: !!checked,
      disabled: !!disabled,
    };
  }

  /**
   * 将 CSS 长度单位减去相应数值并转换
   * 如 getMinusedSize(1) => 1px
   * 如 getMinusedSize(3, 1) => 2px
   * @param {*} size
   * @param {*} minusNum
   */
  static getMinusedSize(size, minusNum = 0) {
    if (typeof size === 'number') {
      return `${size - minusNum}px`;
    }
    if (typeof size !== 'string') {
      return JSON.stringify(size);
    }
    const reg = /\d+/;
    const transedSize = size.replace(reg, matchNum => Number(matchNum) - minusNum);
    // 如果输入了非数字字符串，则直接 return
    if (!/^\d+$/.test(size)) {
      return transedSize;
    }
    return `${transedSize}px`;
  }

  onClick(e) {
    /* eslint-disable prefer-const */
    let { checked, disabled } = this.state;
    const { onChange } = this.props;
    if (disabled) {
      return;
    }
    checked = !checked;
    if (onChange) onChange(checked, e);
    /* eslint-enable prefer-const */
    this.setState({
      checked,
    });
  }

  onKeyPress(e) {
    if (e.keyCode === 13) {
      e.preventDefault();
      this.onClick(e);
    }
  }

  render() {
    const {
      prefixCls, checkedColor,
      name, color, width, height,
    } = this.props;
    const { checked, disabled } = this.state;

    const transedWidth = Switch.getMinusedSize(width);
    const transedHeight = Switch.getMinusedSize(height);
    const minusedHeight = Switch.getMinusedSize(height, 2);
    const borderRadius = transedHeight;

    const divWrapCSS = classnames({
      [`${prefixCls}-switch--wrap`]: true,
      isDisabled: disabled,
    });

    const divWrapStyle = {
      backgroundColor: checked ? checkedColor : color,
      width: transedWidth,
      height: transedHeight,
      borderRadius,
    };

    const divInnerStyle = {
      width: minusedHeight,
      height: minusedHeight,
      borderRadius: minusedHeight,
      border: checked ? `1px solid ${checkedColor}` : `1px solid ${color}`,
      left: !checked ? '1px' : '100%',
      marginLeft: !checked ? 0 : `-${Switch.getMinusedSize(height, 1)}`,
    };

    return (
      <div
        className={divWrapCSS}
        style={divWrapStyle}
        role="switch"
        aria-checked={checked}
        tabIndex="0"
        onClick={this.onClick}
        onKeyPress={this.onKeyPress}
      >
        <input type="checkbox" name={name} value={checked} />
        <div style={divInnerStyle} />
      </div>
    );
  }
}

Switch.defaultProps = {
  checked: false,
  disabled: false,
  name: '',
  color: '#ccc',
  checkedColor: '#343c5c',
  width: '50px',
  height: '30px',
  onChange: () => {},
};

Switch.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  name: PropTypes.string,
  color: PropTypes.string,
  checkedColor: PropTypes.string,
  width: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  height: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  onChange: PropTypes.func,
};

export default prefixClsProperty(Switch);
