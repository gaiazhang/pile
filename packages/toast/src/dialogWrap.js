import * as React from 'react';
import ReactDOM from 'react-dom';

const IS_REACT_16 = !!ReactDOM.createPortal;

class DialogWrap extends React.Component {
  componentDidMount() {
    if (this.props.visible) {
      this.componentDidUpdate();
    }
  }

  componentDidUpdate() {
    if (!IS_REACT_16 && this.props.visible) {
      this.renderDialog(this.props.visible);
    }
  }

  componentWillUnmount() {
    if (this.props.visible) {
      if (!IS_REACT_16) {
        this.renderDialog(false);
      } else {
        // TODO for react@16 createPortal animation
        this.removeContainer();
      }
    } else {
      this.removeContainer();
    }
  }

  getContainer() {
    if (!this.container) {
      const container = document.createElement('div');
      const containerId = `pile-container-${(new Date().getTime())}`;
      container.setAttribute('id', containerId);
      document.body.appendChild(container);
      this.container = container;
    }
    return this.container;
  }

  getComponent() {
    const props = this.props;

    return React.cloneElement(props.children, {
      onAnimateLeave: this.removeContainer.bind(this), ...props,
    });
  }

  removeContainer() {
    if (this.container) {
      if (!IS_REACT_16) {
        ReactDOM.unmountComponentAtNode(this.container);
      }
      this.container.parentNode.removeChild(this.container);
      this.container = null;
      const { onClose } = this.props;
      if (onClose) {
        onClose();
      }
    }
  }

  renderDialog() {
    ReactDOM.unstable_renderSubtreeIntoContainer(
      this,
      this.getComponent(),
      this.getContainer(),
    );
  }

  render() {
    const { visible } = this.props;
    if (IS_REACT_16 && visible) {
      return ReactDOM.createPortal(this.getComponent(), this.getContainer());
    }
    return null;
  }
}

export default DialogWrap;
