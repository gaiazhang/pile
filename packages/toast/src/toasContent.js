import * as React from 'react';
import PropTypes from 'prop-types';
import Icon from '@pile/icon';

class ToastContent extends React.Component {
  componentDidMount() {
    this.startCloseTimer();
  }

  componentDidUpdate(prevProps) {
    const { duration, content, iconType } = this.props;
    if (duration !== prevProps.duration
      || content !== prevProps.content || iconType !== prevProps.iconType) {
      this.restartCloseTimer();
    }
  }

  componentWillUnmount() {
    this.clearCloseTimer();
  }

  close() {
    this.clearCloseTimer();
    // 执行关闭操作
    const { onAnimateLeave } = this.props;
    onAnimateLeave();
  }

  startCloseTimer() {
    const { duration } = this.props;
    if (duration) {
      this.closeTimer = setTimeout(() => {
        this.close();
      }, duration * 1000);
    }
  }

  clearCloseTimer() {
    if (this.closeTimer) {
      clearTimeout(this.closeTimer);
      this.closeTimer = null;
    }
  }

  restartCloseTimer() {
    this.clearCloseTimer();
    this.startCloseTimer();
  }

  render() {
    /* eslint-disable */
    let { content, iconType } = this.props;
    let type = '';

    switch (iconType) {
      case 'success':
        type = 'check-circle';
        break;
      case 'fail':
        type = 'error-circle';
        break;
      case 'load':
        type = 'loading';
        content = '加载中...';
        break;
      case 'warn':
        type = 'warnning-solid-circle';
        break;
      default:
        type = '';
    }

    return (
      <div className="pile-toast">
        {iconType && type ? (<Icon type={type} size="large" />) : null }
        <div className="pile-toast-content" ref={(t) => { this.toastContent = t; }}>
          {content}
        </div>
      </div>
    );
  }
}

ToastContent.propTypes = {
  content: PropTypes.string,
  iconType: PropTypes.oneOf(['success', 'fail', 'load', 'warn', '']),
  duration: PropTypes.number,
  onAnimateLeave: PropTypes.func,
};

ToastContent.defaultProps = {
  content: '',
  iconType: '',
  duration: 3,
  onAnimateLeave: null,
};

export default ToastContent;
