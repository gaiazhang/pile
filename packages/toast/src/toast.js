import * as React from 'react';
import ReactDOM from 'react-dom';
import ToastContent from './toasContent';

let messageInstance;
const getMessageInstance = function createMessage(properties) {
  const div = document.createElement('div');

  document.body.appendChild(div);

  function destroy() {
    ReactDOM.unmountComponentAtNode(div);
    div.parentNode.removeChild(div);
  }

  ReactDOM.render(<ToastContent {...properties} onAnimateLeave={destroy} />, div);

  return { destroy };
};

export default {
  show(properties) {
    if (messageInstance) {
      messageInstance.destroy();
      messageInstance = null;
    }

    messageInstance = getMessageInstance(properties);
  },
  hide() {
    if (messageInstance) {
      messageInstance.destroy();
      messageInstance = null;
    }
  },
};
