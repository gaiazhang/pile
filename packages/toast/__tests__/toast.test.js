import { render } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import { Toast } from '../src';

describe('test toast component', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn(element => element);
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });
  it('snapshot ui', () => {
    const wrapper = render(<Toast content="hi,you" iconType="success" visible />);
    expect(wrapper).toMatchSnapshot();
  });
});
