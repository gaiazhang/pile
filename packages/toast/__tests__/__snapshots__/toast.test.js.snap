// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test toast component snapshot ui 1`] = `
<div
  class="pile-toast"
>
  <i
    class="pile-icon-check-circle large"
  />
  <div
    class="pile-toast-content"
  >
    hi,you
  </div>
</div>
`;
